plugins/pixart-rf/fu-pxi-common.c

```diff
@@ -123,3 +123,18 @@ fu_pxi_composite_receiver_cmd(guint8 opcode,
 	g_byte_array_prepend(wireless_mod_cmd, &rid, 0x01);
 	return TRUE;
 }
+
+gchar *
+fu_pxi_hpac_version_info_parse(const guint16 *hpac_ver)
+{
+	g_autoptr(GByteArray) hpac_ver_buf = g_byte_array_new();
+	fu_byte_array_append_uint8(hpac_ver_buf, '0' + (*hpac_ver / 10000));
+	fu_byte_array_append_uint8(hpac_ver_buf, '0' + ((*hpac_ver / 1000) % 10));
+	fu_byte_array_append_uint8(hpac_ver_buf, 0x2e);
+	fu_byte_array_append_uint8(hpac_ver_buf, '0' + ((*hpac_ver / 100) % 10));
+	fu_byte_array_append_uint8(hpac_ver_buf, '0' + ((*hpac_ver / 10) % 10));
+	fu_byte_array_append_uint8(hpac_ver_buf, 0x2e);
+	fu_byte_array_append_uint8(hpac_ver_buf, 0x30);
+	fu_byte_array_append_uint8(hpac_ver_buf, '0' + (*hpac_ver % 10));
+	return g_strndup((gchar *)hpac_ver_buf->data, hpac_ver_buf->len);
```

Comment:
I think something like `return g_strdup_printf("%u.%u.%u", hpac_ver.....` is much more obvious -- no?

Reply:
Sorry, May I ask how to use "g_strdup_printf" function ?

The range of uint16 hpac_ver is 0~55539, it means 55,53,9. I need to convert to "55,53,9" from number 55539.

I just tried g_strdup_printf("%u,%u,%u",hpac_ver), it shows "Current version:    191,1,49"
But hpac_ver is 191, so it's should be "Current version:    00.19.01"

Reply:
Does something like `return g_strdup_printf("%u%u,%u,%u", hpac_ver / 10000, (hpac_ver / 1000) % 10, ((hpac_ver / 100) % 10), (hpac_ver / 10) % 10, hpac_ver % 10);` work?

Reply:
@hughsie  Thanks for your help! it's working now. And I have committed the newer one to fix it.